lib/eco/eco.py

```diff
@@ -1042,5 +1064,49 @@ def run(self):
             for line in iter(p.stdout.readline, b''):
                 self.emit(self.signal, line.rstrip())
 
+
+class ProfileThread(QThread):
+    def __init__(self, window, parent):
+        QThread.__init__(self, parent=parent)
+        self.window = window
+        self.signal_done = QtCore.SIGNAL("finished")
+        self.signal = QtCore.SIGNAL("output")
+
+    def run(self):
+        p = self.window.getEditor().tm.export(profile=True)
+        # Using read() here, rather than readline() because profiler output
+        # often includes blank lines in the middle of the output.
+        if p:
+            text = p.stdout.read()
+            self.emit(self.signal, text.strip())
+        self.emit(self.signal_done, None)
+
+class Throbber(QLabel):
+    """Throbber which displays in the left-hand corner of the tabbed notebook.
+    Used to alert the user that a potentially long-running background
+    profile is taking place.
+    """
+    def __init__(self, tab_bar):
+        super(Throbber, self).__init__()
+        self.tab_bar = tab_bar
+        self.setAlignment(Qt.AlignCenter)
+        self._movie = QMovie("gui/throbber.gif")
+        self.setMovie(self._movie)
+
+    def hide(self):
+        """Hide throbber in left hand corner of tabbed notebook.
+        """
+        self._movie.stop()
+        self.tab_bar.setCornerWidget(None, corner=Qt.TopLeftCorner)
```

Comment:
It doesn't matter a great deal, but I think the waiting animation (which I quite like by the way) looks a bit better on the top right corner. What do you think?